### PR TITLE
Ui

### LINKED ISSUE #1: Health meter
Create a health web component that takes in a character's health and name and displays it on the screen wherever we decide to place it (i.e. top bar for player or info section for enemy) 

needs to rerender when characters hp mutates.

handle cleanup of ui in character's destroy method that is run when character's hp reaches 0.

### LINKED ISSUE #17: Info Panel
Create a web component that can display the following:

**Enemy alive state:**
The name of the enemy
The enemy's current health
Any effects/timers connected to the enemy 
A button that allows player to attack the enemy

**Enemy dead state:**
Mimic **chest opened state**

**Chest unopened state:**
Name of chest type (1 standard chest type for now)
Button that allows the player to open the chest

**Chest opened state:** 
Grid of all items inside the chest
button to take all items
clicking an item clears contents and displays **item state**
this version of item state has an additional button to go back to chest state

**Item state:**
item name
item stats
button to take item
if equippable: button to equip
if useable: button to use

### PR DESCRIPTION
**Pull request by KevinW90**
resolves #17 
resolves #1 

-player health meter
-enemy health meter

other notes
-stopped using Player from global space
-stopped using Room from global space
-grabbed Enemies from the current room
-added UI object to store and retrieve UI elements
-changed player menu to a component
-added player menu component to UI
-added push to add items to chest
-open chest button displays item names (need icons)

### CODE REVIEW

File: js/classes/enterRoom.js [hunk 1]
```diff
@@ -59,10 +56,19 @@ class CurrentRoom {
...
+    console.log(this._contents);
+
+    // TODO: remove this before pushing to prod
+    window.Room = this;
+
+    // setTimeout(() => {
+    //   this._poison = generateEffect(poison, this._player);
+    //   this._player.effects.push(this._poison);
+    // }, 10000);
+  }
```
Comment (heykc) [inline js/classes/enterRoom.js:68]: when you comment a block out, just toss a comment above it explaining why. No need to change this now, but for future reference

File: js/classes/player.js [hunk 2]
```diff
@@ -37,22 +36,71 @@ class Player extends Character {
...
+    
+    this._elements['playerMenu'].appendChild(buildElement('touch-icon',
+                                                          {class: 'menu-item'},
+                                                          {icon: 'boxes',
+                                                           type: 'inv'}));
+    this._elements['playerMenu'].appendChild(buildElement('touch-icon', 
+                                                          {class: 'menu-item'}, 
+                                                          {icon: 'arrows-alt',
+                                                           type: 'move'}));
+
```
Comment (heykc) [inline js/classes/player.js:65]: this is some strange formatting. could you clean this up before the merge?

Comment (KevinW90) [reply js/classes/player.js:65]: I don't know how else to format this, but I did change playermenu to a component so it isn't as long any more. Trying to get across that buildElement takes some args. Is there a suggestion for cleaning this formatting?

File: js/utils/ui.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+export const UI = {
+  infoSection: document.querySelector('#info-section'),
+  iconBar: document.querySelector('#icon-bar'),
+  playerInfo: document.querySelector('#player-info'),
+  playerMenu: buildElement(
+    'player-menu',
+    null,
+    {isOpen: true} //because render flips the state
+  )
+}
```
Comment (heykc) [inline js/utils/ui.js:14]: this is an interesting module, next time we meet up can you talk me through how this works?

**Pull request merged**